Sources/MongoSwift/BSON/AnyBSONValue.swift

```diff
@@ -2,8 +2,19 @@ import Foundation
 
 /// A struct wrapping a `BSONValue` type that allows for encoding/
 /// decoding `BSONValue`s of unknown type.  
-public struct AnyBSONValue: Codable, Equatable {
-    /// The `BSONValue` wrapped by this struct. 
+public struct AnyBSONValue: Codable, Equatable, Hashable {
+    // swiftlint:disable:next legacy_hashing
+    public var hashValue: Int {
+        if let doc = self.value as? Document {
+            return doc.extendedJSON.hashValue
+        } else {
+            let doc: Document = ["value": self.value]
+            // need to add some string to the beginning to ensure no collisions with the document case.
```

Comment:
am I understanding correctly that the problem here is that the hash value for `AnyBSONValue(["value": 1] as Document)` would match the hash value for `AnyBSONValue(1)`?

Reply:
Yep. Since `<any string> + <valid extended JSON>` will never be valid extended JSON, it helps avoid collisions.